File: internal/models/monitoring/prometheus.go
package monitoring

import "time"

// PrometheusInstance Prometheus 数据源实例
type PrometheusInstance struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Name      string    `gorm:"size:100;not null;uniqueIndex" json:"name"`
	URL       string    `gorm:"size:500;not null" json:"url"` // e.g. http://prometheus:9090
	AuthType  string    `gorm:"size:20;default:'none'" json:"auth_type"` // none / basic / bearer
	Username  string    `gorm:"size:200" json:"username"`
	Password  string    `gorm:"size:500" json:"password"` // 加密存储(basic auth) or bearer token
	IsDefault bool      `gorm:"default:false" json:"is_default"`
	Status    string    `gorm:"size:20;default:'active'" json:"status"`
	CreatedBy *uint     `json:"created_by"`
}

func (PrometheusInstance) TableName() string { return "prometheus_instances" }

File: web/src/services/prometheus.ts
import request from './api'
import type { ApiResponse } from '../types'

export interface PrometheusInstance {
  id?: number
  created_at?: string
  updated_at?: string
  name: string
  url: string
  auth_type: string
  username: string
  password: string
  is_default: boolean
  status: string
  created_by?: number
}

export const prometheusApi = {
  // Instance CRUD
  listInstances: () =>
    request.get<ApiResponse<PrometheusInstance[]>>('/prometheus/instances'),
  getInstance: (id: number) =>
    request.get<ApiResponse<PrometheusInstance>>(`/prometheus/instances/${id}`),
  createInstance: (data: Partial<PrometheusInstance>) =>
    request.post<ApiResponse<PrometheusInstance>>('/prometheus/instances', data),
  updateInstance: (id: number, data: Partial<PrometheusInstance>) =>
    request.put<ApiResponse<PrometheusInstance>>(`/prometheus/instances/${id}`, data),
  deleteInstance: (id: number) =>
    request.delete<ApiResponse<null>>(`/prometheus/instances/${id}`),
  testConnection: (id: number) =>
    request.post<ApiResponse<null>>(`/prometheus/instances/${id}/test`),

  // Query Proxy BFF
  query: (params: { query: string; time?: string; instance_id?: number }) =>
    request.get<ApiResponse<any>>('/prometheus/query', { params }),
  queryRange: (params: { query: string; start: string; end: string; step?: string; instance_id?: number }) =>
    request.get<ApiResponse<any>>('/prometheus/query_range', { params }),
  labels: (params?: { instance_id?: number }) =>
    request.get<ApiResponse<string[]>>('/prometheus/labels', { params }),
  labelValues: (name: string, params?: { instance_id?: number }) =>
    request.get<ApiResponse<string[]>>(`/prometheus/label/${name}/values`, { params }),
  series: (params: { match: string; start?: string; end?: string; instance_id?: number }) =>
    request.get<ApiResponse<any>>('/prometheus/series', { params }),
  targets: (params?: { instance_id?: number }) =>
    request.get<ApiResponse<any>>('/prometheus/targets', { params }),
}
